display_list/benchmarking/dl_builder_benchmarks.cc

```diff
@@ -220,6 +220,43 @@ static void BM_DisplayListDispatchDefault(
   }
 }
 
+static void BM_DisplayListDispatchByIndexDefault(
+    benchmark::State& state,
+    DisplayListDispatchBenchmarkType type) {
+  bool prepare_rtree = NeedPrepareRTree(type);
+  DisplayListBuilder builder(prepare_rtree);
+  for (int i = 0; i < 5; i++) {
+    InvokeAllOps(builder);
+  }
+  auto display_list = builder.Build();
+  DlOpReceiverIgnore receiver;
+  while (state.KeepRunning()) {
+    DlIndex end = display_list->record_count();
+    for (DlIndex i = 0u; i < end; i++) {
+      display_list->Dispatch(receiver, i);
+    }
```

Comment:
So this is really the meat of what this change allows: dispatch takes an offset and so I can choose to dispatch at any particular offset.

The tricky part is the op kinds right? We'd still need to manage things like paint state, clip state, transform state and distinguish between them when re-dispatching, correct?

Reply:
That's where GetOpCategory comes into play. You can easily figure out if it is a Save/Restore or a Rendering op or an Attribute/Transform/Clip op.

Reply:
With the latest change you can also do `for (DlIndex i : *display_list)` - woot...